File: Bronze1-1/06-21-am/unit03/duplicates-I/duplicates.cpp
/*
 * Duplicates
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int Q; cin >> Q; //# of queries
	while(Q--) {
		int N, K;
		cin >> N >> K;
	
		int A[N];
		for(int j=0; j<N; j++)
			cin >> A[j];
		
		int freq[1001];
		fill(freq, freq+1001, 0);
		for(int j=0; j<N; j++)
			freq[ A[j] ]++;
		
		bool exist = false;
		for(int x=0; x<=1000; x++) {
			if(freq[x]>=K) {
				exist = true;
				break;
			}
		}
		cout << (exist ? "YES" : "NO") << '\n';
	}
}

File: Bronze1-1/06-21-am/unit03/duplicates-I/duplicates.java
/*
 * Duplicates
 */
import java.util.*;
public class duplicates {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int Q = in.nextInt(); //# of queries
		while(Q-->0) {
			int N = in.nextInt();
			int K = in.nextInt();
			
			int[] A = new int[N];
			for(int j=0; j<N; j++)
				A[j] = in.nextInt();
			
			int[] freq = new int[1001];
			for(int j=0; j<N; j++)
				freq[ A[j] ]++;
			
			boolean exist = false;
			for(int x=0; x<=1000; x++) {
				if(freq[x]>=K) {
					exist = true;
					break;
				}
			}
			System.out.println(exist ? "YES" : "NO");
		}
		in.close();
	}
}